test/handlers/web/technologiesTest.js

```diff
@@ -0,0 +1,313 @@
+            req.validationErrors.restore(); // get rid of the default stub created in beforeEach()
+            sinon.stub(req, 'validationErrors').returns(true);
+
+            webTechnologies.getVotes(req, res);
+
+            sinon.assert.notCalled(res.render);
+        });
+
+        it("should pass the technology object to the rendered view", function() {
+            webTechnologies.getVotes(req, res);
+
+            sinon.assert.calledOnce(res.render);
+            expect(res.render.args[0][1]).to.contain({'technology': technologyFromDb});
+        });
+    });
+
+    describe("updateStatus", function() {
+        it("should render updateStatus page", function() {
+            webTechnologies.updateStatus(req, res);
+
+            sinon.assert.calledOnce(res.render);
+            expect(res.render.args[0][0]).that.is.a('string').to.contain('updateStatus');
+        });
+
+        it("should NOT render updateStatus page if validation failed", function() {
+            req.validationErrors.restore(); // get rid of the default stub created in beforeEach()
+            sinon.stub(req, 'validationErrors').returns(true);
+
+            webTechnologies.updateStatus(req, res);
+
+            sinon.assert.notCalled(res.render);
+        });
+
+        it("should pass the technology object to the rendered view", function() {
+            webTechnologies.updateStatus(req, res);
+
+            sinon.assert.calledOnce(res.render);
+            expect(res.render.args[0][1]).to.contain({'technology': technologyFromDb});
+        });
+    });
+
+    describe("addProject", function() {
+        beforeEach(function() {
+            sinon.stub(projectsDao, 'getAllForTechnology', function(techid, cb){
+                cb({});
+            });
+            sinon.stub(projectsDao, 'getAll', function(cb){
+                cb({filter: sinon.stub()});
+            });
+        });
+
+        afterEach(function() {
+            projectsDao.getAllForTechnology.restore();
+            projectsDao.getAll.restore();
+        });
+
+        it("should render addProject page", function() {
+            webTechnologies.addProject(req, res);
+
+            sinon.assert.calledOnce(res.render);
+            expect(res.render.args[0][0]).that.is.a('string').to.contain('addProject');
+        });
+
+        it("should NOT render addProject page if validation failed", function() {
+            req.validationErrors.restore(); // get rid of the default stub created in beforeEach()
+            sinon.stub(req, 'validationErrors').returns(true);
+
+            webTechnologies.addProject(req, res);
+
+            sinon.assert.notCalled(res.render);
+        });
+
+        it("should pass the technology object to the rendered view", function() {
+            webTechnologies.addProject(req, res);
+
+            sinon.assert.calledOnce(res.render);
+            expect(res.render.args[0][1]).to.contain({'technology': technologyFromDb});
+        });
+    });
```

Comment:
perhaps we could add some tests for project.getAllForTechnology and project.getAll being called and outputs passed to the renderer